components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/dao/impl/SAML2ArtifactInfoDAOImpl.java

```diff
@@ -0,0 +1,192 @@
+import java.sql.SQLException;
+import java.sql.Timestamp;
+
+/**
+ * Default implementation of SAML2ArtifactInfoDAO.
+ */
+public class SAML2ArtifactInfoDAOImpl implements SAML2ArtifactInfoDAO {
+
+    @Override
+    public void storeArtifactInfo(SAML2ArtifactInfo saml2ArtifactInfo) throws ArtifactBindingException {
+
+        final String ARTIFACT_INFO_STORE_SQL = "INSERT INTO IDN_SAML2_ARTIFACT_STORE(SOURCE_ID, " +
+                "MESSAGE_HANDLER, AUTHN_REQ_DTO, SESSION_ID, INIT_TIMESTAMP, EXP_TIMESTAMP, ASSERTION_ID) " +
+                "VALUES (?, ?, ?, ?, ?, ?, ?)";
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+
+        try {
+            jdbcTemplate.executeInsert(ARTIFACT_INFO_STORE_SQL, (preparedStatement -> {
+                preparedStatement.setBytes(1, saml2ArtifactInfo.getSourceId());
+                preparedStatement.setBytes(2, saml2ArtifactInfo.getMessageHandler());
+                try {
+                    setBlobObject(preparedStatement, saml2ArtifactInfo.getAuthnReqDTO(), 3);
+                } catch (IOException e) {
+                    throw new SQLException(e);
+                }
+                preparedStatement.setString(4, saml2ArtifactInfo.getSessionID());
+                preparedStatement.setTimestamp(5, new Timestamp(saml2ArtifactInfo.getInitTimestamp().getMillis()));
+                preparedStatement.setTimestamp(6, new Timestamp(saml2ArtifactInfo.getExpTimestamp().getMillis()));
+                preparedStatement.setString(7, saml2ArtifactInfo.getAssertionID());
+            }), saml2ArtifactInfo, true);
+        } catch (DataAccessException e) {
+            throw new ArtifactBindingException("Error while storing SAML2 artifact information.");
+        }
+    }
+
+    @Override
+    public SAML2ArtifactInfo getSAMLArtifactInfo(byte[] sourceId, byte[] messageHandler) throws ArtifactBindingException {
+
+        final String ARTIFACT_INFO_RETRIEVE_SQL = "SELECT ID, AUTHN_REQ_DTO, SESSION_ID, INIT_TIMESTAMP, " +
+                "EXP_TIMESTAMP FROM IDN_SAML2_ARTIFACT_STORE WHERE SOURCE_ID=? AND MESSAGE_HANDLER=?";
+        final String ASSERTION_DELETE_SQL = "DELETE FROM IDN_SAML2_ARTIFACT_STORE WHERE ID=?";
+        SAML2ArtifactInfo saml2ArtifactInfo;
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+
+        try {
+            saml2ArtifactInfo = jdbcTemplate.fetchSingleRecord(ARTIFACT_INFO_RETRIEVE_SQL, (resultSet, rowNumber) ->
+                    {
+                        try {
+                            return new SAML2ArtifactInfo(resultSet.getInt(1),
+                                    (SAMLSSOAuthnReqDTO) getBlobObject(resultSet.getBinaryStream(2)),
+                                    resultSet.getString(3),
+                                    new DateTime(resultSet.getTimestamp(4)),
+                                    new DateTime(resultSet.getTimestamp(5)));
+                        } catch (IOException | ClassNotFoundException e) {
```

Comment:
Better to catch specific exception and throw the error with information.
e.g.
catch (IOException e) {
throw new SQLException("Error in reading the blob from the database for sourceId: messageHandler" + sourceId+":"+ messageHandler, e);
} catch (ClassNotFoundException e) {
throw new SQLException("Unable to deserialize the object from blob..for sourceId: messageHandler" + sourceId+":"+ messageHandler, e);
}

Reply:
Can we print the sourceID and messageHandler here since they are byte arrays?